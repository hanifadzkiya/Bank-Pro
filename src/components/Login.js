import React, { Component } from "react";
import {Redirect} from 'react-router-dom';

class Login extends Component {
    constructor(props){
        super(props)
        this.login = this.login.bind(this);
    }

    login(){
        sessionStorage.setItem('authToken', 'testLoginToken');
    }

    render() {
        return (
            <div>
                <h2>Login</h2>
                <p>Tekan tombol di bawah untuk login</p>
                <form onSubmit = {this.login}>
                    <input type="submit" value="Login"/>
                </form>
            </div>
        );
    }
}
export default Login;