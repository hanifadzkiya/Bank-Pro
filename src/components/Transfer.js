import React, { Component } from "react";
import { callTransactionService } from "../services/PostRequest";

class Transfer extends Component {
    constructor(props){
        super(props)
        this.onChange = this.onChange.bind(this);
        this.state = {
            NomorPenerima: '',
            Nominal: 0
        };
    }

    onChange = (event) =>{
        event.preventDefault();
        this.setState({
            [event.target.name] : event.target.value
        })
    }

    render() {
        return (
            <div>
                <h2>Transfer</h2>
                <p>Pengguna dapat mentransfer uang ke rekening lain di Bank Pro.
                    Tidak ada potongan tambahan ketika mentransfer ke rekening bank lain.
                    Saat transaksi selesai, ada pesan berhasil atau gagal yang muncul.</p>
                <div className="Transfer">
                    <div className="Home-body">
                        <form onSubmit = {callTransactionService(sessionStorage.getItem('rekening'),this.state.NomorPenerima,this.state.Nominal)}>
                            <input
                                name="NomorPenerima"
                                type="text"
                                onChange={this.onChange}
                            /><br></br>
                            <input
                                name="Nominal"
                                type="number"
                                onChange={this.onChange}
                            /><br></br>
                            <input type="submit" value="Transfer"/>
                        </form>
                    </div>
                </div>
            </div>
        );
    }
}

export default Transfer;
