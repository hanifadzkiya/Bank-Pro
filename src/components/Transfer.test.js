import React from "react";
import Transfer from './Transfer';

import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

import { shallow, mount } from 'enzyme';
import sinon from 'sinon';
import Login from "./Login";

it('Render Transfer', () => {
    const wrapper = shallow(<Transfer />);
    expect(wrapper.contains(<h2>Transfer</h2>)).toEqual(true);
    expect(wrapper.contains(<p>Pengguna dapat mentransfer uang ke rekening lain di Bank Pro.
        Tidak ada potongan tambahan ketika mentransfer ke rekening bank lain.
        Saat transaksi selesai, ada pesan berhasil atau gagal yang muncul.</p>)).toEqual(true);
    expect(wrapper).toMatchSnapshot();
})


it('check form',()=>
{
    const preventDefault = jest.fn();
    const wrapper = shallow(<Transfer/>);
    wrapper.find('input[type="number"]').simulate('change', {preventDefault, target: {value: 1000,name: 'Nominal'}});
    expect(wrapper.state('Nominal')).toEqual(1000);
});

it('check form2',()=>
{
    const preventDefault = jest.fn();
    const wrapper = shallow(<Transfer/>);
    wrapper.find('input[type="text"]').simulate('change', {preventDefault, target: {value: '1000',name: 'NomorPenerima'}});
    expect(wrapper.state('NomorPenerima')).toEqual('1000');
});