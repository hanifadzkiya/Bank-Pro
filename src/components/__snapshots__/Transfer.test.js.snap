// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Transfer 1`] = `ShallowWrapper {}`;
